README.md
# TYPO3 Extension bib

Bib is a publication reference (bibliography) manager plugin for
TYPO3. It provides a browseable list view for publication references
and a convenient publication reference editor.

Bib is forked from the Sevenpack extension by Sebastian Holtermann.

## Features:

* A publication reference database integrated into the TYPO3 framework.
* A heavily customizable frontend list view for publication references.
* A frontend single view (disabled by default).
* An optional author browser bar
* An optional search bar
* An optional preferences bar
* A convenient frontend editor for publication reference
* Optional PDF text caching for full text search purposes
* Export of the reference database to BibTeX or a homebrewn Bib XML format.
* Import of references from a BibTeX file or a Bib XML file.
* Localizations for English, German and Italian (very outdated).

## Documentation

The documentation can be found at [https://github.com/subugoe/typo3-bib/wiki](https://github.com/subugoe/typo3-bib/wiki).


ChangeLog
2011-01-04      Sebastian Holtermann  <sebholt@web.de>
	* Editor: Basic support for per publication edit restrictions (K. Weidenbach)
	* Code: Small space at end of line cleanups
	* Release version 0.7.16

2010-12-20      Sebastian Holtermann  <sebholt@web.de>
	* Backend: Fixed authorship label (R. Wildling)
	* Code: Indentation cleanups
	* Release version 0.7.15

2010-12-06      Sebastian Holtermann  <sebholt@web.de>
	* SQL generation fix (T. Schaefer)
	* German localization fixes (T. Schaefer)
	* More german localization fixes
	* Release version 0.7.14

2010-10-08      Sebastian Holtermann  <sebholt@web.de>
	* Backend: The DOI field now appears in the backend, too. (T. Schaefer)
	* Release version 0.7.13

2010-09-23      Sebastian Holtermann  <sebholt@web.de>
	* General: Changed <? to <?php in res/class.tx_sevenpack_labels.php
	* General: Some small fixes (T. Schaefer)
	* Release version 0.7.12

2010-09-16      Sebastian Holtermann  <sebholt@web.de>
	* Editor: Better handling for no_edit and no_show flags (C. Lerrahn)
	* Editor: Dicarding of no_edit and no_show fields when saving
	* Editor: Check for invalid pids when saving (C. Lerrahn)
	* Editor: General code overhauling
	* Import: Check for invalid pids when saving
	* General: Typo fixes (T. Schaefer)
	* Release version 0.7.11

2010-08-17      Sebastian Holtermann  <sebholt@web.de>
	* Editor: Fields can be hidden entirely with the new no_show variable
	* Import: BibTeX publications with an unknown bitype don't trigger invalid double imports anymore
	* ChangeLog: Fixed some dates which had a wrong order ;-)
	* Release version 0.7.10

2010-01-21      Sebastian Holtermann  <sebholt@web.de>
	* List: Fixed double reference count in the year navigation view when *all years* was selected (O. Fischer)
	* List: Fixed the editor default year for new entries if *all years* was selected
	* Internals: Separated reference reader and writer classes - May give a little more speed
	* Release version 0.7.9

2009-11-05      Sebastian Holtermann  <sebholt@web.de>
	* List: Fixed broken link creation when file_url contained ampersand characters
	* List: PHP cleanups and optimisations
	* Release version 0.7.8

2009-11-04      Sebastian Holtermann  <sebholt@web.de>
	* Full text cache: Proper shell argument escaping for pdftotext
	* Documentation: Updated PDF documentation
	* Release version 0.7.7

2009-11-04      Sebastian Holtermann  <sebholt@web.de>
	* Full text cache: Time and number limit for full text cache updates.
	                   This should help to avoid hitting a PHP CPU/Time limit.
	* Full text cache: Check if a temporary file could be created
	* Localization: Fixed english link titles in the year navigation bar
	* Release version 0.7.6

2009-10-30      Sebastian Holtermann  <sebholt@web.de>
	* General: The full text cache remembers the filename of the
	           cached file to avoid unneccessary cache updates
	* Database: New full_text_file_url field (not editable)
	* Release version 0.7.5

2009-10-29      Sebastian Holtermann  <sebholt@web.de>
	* List: New tags filter option
	* Release version 0.7.4

2009-09-28      Sebastian Holtermann  <sebholt@web.de>
	* Documentation: Updated with a description of the automatic pdf text extraction
	* BibTeX import/export: Fix for a wrong accent conversion (D. Sauveron).
	* Release version 0.7.3

2009-09-23      Sebastian Holtermann  <sebholt@web.de>
	* Database: The data types of affiliation and booktitle are changed 
	            from tinytext to text which is longer (D. Sauveron).
	* Editor: The input fields for affiliation and booktitle are bigger.
	* Release version 0.7.2

2009-08-14      Sebastian Holtermann  <sebholt@web.de>
	* General: If no storage is set the current page will be used as storage
	* General: Normal pages can be storages even from the BE (C. Metter)
	* Editor: Fixed not appearing "Add" icon
	* Documentation: Updated PDF documentation
	* Release version 0.7.1

2009-07-01      Sebastian Holtermann  <sebholt@web.de>
	* General: New single view (disabled by default)
	* List: Year navigation bar starts with the _current_ year instead of with _all_ years
	* Release version 0.7

2009-03-06      Sebastian Holtermann  <sebholt@web.de>
	* List: New search bar
	* List: New author browser bar
	* List: New preferences bar
	* List: Better year navigation bar
	* List: New enumeration type: File-Icon
	* List: Fields can be restricted to FE user groups
	* List: The HTML template is splitted into multiple files
	* List: Style changes
	* Export: Export can be restricted to FE user groups
	* Editor: Field labels can change with the bibtype
	* Database: PDF full text can be cached and searched by the search bar
	* Database: New Publication types: Thesis, Manuscript, Report
	* Database: New Fields: tags, web_url, web_url2, misc2, 

tca.php
<?php
if (!defined ('TYPO3_MODE')) 	die ('Access denied.');

$TCA['tx_sevenpack_references'] = array (
	'ctrl' => $TCA['tx_sevenpack_references']['ctrl'],
	'interface' => array (
		'showRecordFieldList' => $TCA['tx_sevenpack_references']['feInterface']['fe_admin_fieldList']
	),
	'feInterface' => $TCA['tx_sevenpack_references']['feInterface'],
	'columns' => array (
		'hidden' => array (
#			'exclude' => 1,
			'label'   => 'LLL:EXT:lang/locallang_general.xml:LGL.hidden',
			'config'  => array (
				'type'    => 'check',
				'default' => '0'
			)
		),
		'bibtype' => Array (
#			'exclude' => 1,
			'label' => 'LLL:EXT:sevenpack/locallang_db.xml:tx_sevenpack_references_bibtype',
			'config' => Array (
				'type' => 'select',
				'items' => Array (
					Array('LLL:EXT:sevenpack/locallang_db.xml:tx_sevenpack_references_bibtype_I_0', '0'),
					Array('LLL:EXT:sevenpack/locallang_db.xml:tx_sevenpack_references_bibtype_I_1', '1'),
					Array('LLL:EXT:sevenpack/locallang_db.xml:tx_sevenpack_references_bibtype_I_2', '2'),
					Array('LLL:EXT:sevenpack/locallang_db.xml:tx_sevenpack_references_bibtype_I_3', '3'),
					Array('LLL:EXT:sevenpack/locallang_db.xml:tx_sevenpack_references_bibtype_I_4', '4'),
					Array('LLL:EXT:sevenpack/locallang_db.xml:tx_sevenpack_references_bibtype_I_5', '5'),
					Array('LLL:EXT:sevenpack/locallang_db.xml:tx_sevenpack_references_bibtype_I_6', '6'),
					Array('LLL:EXT:sevenpack/locallang_db.xml:tx_sevenpack_references_bibtype_I_7', '7'),
					Array('LLL:EXT:sevenpack/locallang_db.xml:tx_sevenpack_references_bibtype_I_8', '8'),
					Array('LLL:EXT:sevenpack/locallang_db.xml:tx_sevenpack_references_bibtype_I_9', '9'),
					Array('LLL:EXT:sevenpack/locallang_db.xml:tx_sevenpack_references_bibtype_I_10', '10'),
					Array('LLL:EXT:sevenpack/locallang_db.xml:tx_sevenpack_references_bibtype_I_11', '11'),
					Array('LLL:EXT:sevenpack/locallang_db.xml:tx_sevenpack_references_bibtype_I_12', '12'),
					Array('LLL:EXT:sevenpack/locallang_db.xml:tx_sevenpack_references_bibtype_I_13', '13'),
					Array('LLL:EXT:sevenpack/locallang_db.xml:tx_sevenpack_references_bibtype_I_14', '14'),
					Array('LLL:EXT:sevenpack/locallang_db.xml:tx_sevenpack_references_bibtype_I_15', '15'),
					Array('LLL:EXT:sevenpack/locallang_db.xml:tx_sevenpack_references_bibtype_I_16', '16'),
					Array('LLL:EXT:sevenpack/locallang_db.xml:tx_sevenpack_references_bibtype_I_17', '17'),
					Array('LLL:EXT:sevenpack/locallang_db.xml:tx_sevenpack_references_bibtype_I_18', '18'),
					Array('LLL:EXT:sevenpack/locallang_db.xml:tx_sevenpack_references_bibtype_I_19', '19'),
				),
				'size' => 1,
				'maxitems' => 1,
			)
		),
		'citeid' => Array (
#			'exclude' => 1,
			'label' => 'LLL:EXT:sevenpack/locallang_db.xml:tx_sevenpack_references_citeid',
			'config' => Array (
				'type' => 'input',
				'size' => '48',
				'max' => '255',
				'eval' => 'nospace,uniqueInPid',
			)
		),
		'title' => Array (
#			'exclude' => 1,
			'label' => 'LLL:EXT:sevenpack/locallang_db.xml:tx_sevenpack_references_title',
			'config' => Array (
				'type' => 'text',
				'cols' => '48',
				'rows' => '3',
			)
		),
		'journal' => Array (
#			'exclude' => 1,
			'label' => 'LLL:EXT:sevenpack/locallang_db.xml:tx_sevenpack_references_journal',
			'config' => Array (
				'type' => 'input',
				'size' => '48',
				'max'  => '255',
				'eval' => 'trim',
			)
		),
		'year' => Array (
#			'exclude' => 1,
			'label' => 'LLL:EXT:sevenpack/locallang_db.xml:tx_sevenpack_references_year',
			'config' => Array (
				'type'     => 'input',
				'size'     => '4',
				'max'      => '4',
				'eval'     => 'int',
				'range'    => Array (
					'upper' => '10000',
					'lower' => '0'
				),
				'default' => 0
			)
		),
		'volume' => Array (
#			'exclude' => 1,
			'label' => 'LLL:EXT:sevenpack/locallang_db.xml:tx_sevenpack_references_volume',
			'config' => Array (
				'type' => 'input',
				'size' => '48',
				'max' => '255',
				'eval' => 'trim',
			)
		),
		'number' => Array (
#			'exclude' => 1,
			'label' => 'LLL:EXT:sevenpack/locallang_db.xml:tx_sevenpack_references_number',
			'config' => Array (
				'type' => 'input',
				'size' => '48',
				'max' => '255',
				'eval' => 'trim',
			)
		),
		'number2' => Array (
#			'exclude' => 1,
			'label' => 'LLL:EXT:sevenpack/locallang_db.xml:tx_sevenpack_references_number2',
			'config' => Array (
				'type' => 'input',
				'size' => '48',
				'max' => '255',
				'eval' => 'trim',
			)
		),
		'pages' => Array (
#			'exclude' => 1,
			'label' => 'LLL:EXT:sevenpack/locallang_db.xml:tx_sevenpack_references_pages',
			'config' => Array (
				'type' => 'input',
				'size' => '48',
				'max' => '255',
				'eval' => 'trim',
			)
		),
		'day' => Array (
#			'exclude' => 1,
			'label' => 'LLL:EXT:sevenpack/locallang_db.xml:tx_sevenpack_references_day',
			'config' => Array (
				'type' => 'select',
				'items' => Array (
					Array('-',   '0'),
					Array('1',   '1'),
					Array('2',   '2'),
					Array('3',   '3'),
					Array('4',   '4'),
					Array('5',   '5'),
					Array('6',   '6'),
					Array('7',   '7'),
					Array('8',   '8'),
					Array('9',   '9'),
					Array('10', '10'),
					Array('11', '11'),
					Array('12', '12'),
					Array('13', '13'),
					Array('14', '14'),
					Array('15', '15'),
					Array('16', '16'),
					Array('17', '17'),
					Array('18', '18'),
					Array('19', '19'),
					Array('20', '20'),
					Array('21', '21'),
					Array('22', '22'),
					Array('23', '23'),
					Array('24', '24'),
					Array('25', '25'),
					Array('26', '26'),
					Array('27', '27'),
					Array('28', '28'),
					Array('29', '29'),
					Array('30', '30'),
					Array('31', '31')
				),
				'size' => 1,
				'maxitems' => 1,
			)
		),
		'month' => Array (
#			'exclude' => 1,
			'label' => 'LLL:EXT:sevenpack/locallang_db.xml:tx_sevenpack_references_month',
			'config' => Array (
				'type' => 'select',
				'items' => Array (
					Array('-',   '0'),
					Array('1',   '1'),
					Array('2',   '2'),
					Array('3',   '3'),
					Array('4',   '4'),
					Array('5',   '5'),
					Array('6',   '6'),
					Array('7',   '7'),
					Array('8',   '8'),
					Array('9',   '9'),
					Array('10', '10'),
					Array('11', '11'),
					Array('12', '12'),
				),
				'size' => 1,
				'maxitems' => 1,
			)
		),
		'abstract' => Array (
#			'exclude' => 1,
			'label' => 'LLL:EXT:sevenpack/locallang_db.xml:tx_sevenpack_references_abstract',
			'config' => Array (
				'type' => 'text',
				'cols' => '48',
				'rows' => '10',
			)
		),
		'affiliation' => Array (
#			'exclude' => 1,
			'label' => 'LLL:EXT:sevenpack/locallang_db.xml:tx_sevenpack_references_affiliation',
			'config' => Array (
				'type' => 'text',
				'cols' => '48',
				'rows' => '2',
				'eval' => 'trim',
			)
		),
		'note' => Array (
#			'exclude' => 1,
			'label' => 'LLL:EXT:sevenpack/locallang_db.xml:tx_sevenpack_references_note',
			'config' => Array (
				'type' => 'text',
				'cols' => '48',
				'rows' => '5',
			)
		),
		'annotation' => Array (
#			'exclude' => 1,
			'label' => 'LLL:EXT:sevenpack/locallang_db.xml:tx_sevenpack_references_annotation',
			'config' => Array (
				'type' => 'text',
				'cols' => '48',
				'rows' => '5',
			)
		),
		'keywords' => Array (
#			'exclude' => 1,
			'label' => 'LLL:EXT:sevenpack/locallang_db.xml:tx_sevenpack_references_keywords',
			'config' => Array (
				'type' => 'text',
				'cols' => '48',
				'rows' => '2',
			)
		),
		'tags' => Array (
#			'exclude' => 1,
			'label' => 'LLL:EXT:sevenpack/locallang_db.xml:tx_sevenpack_references_tags',
			'config' => Array (
				'type' => 'text',
				'cols' => '48',
				'rows' => '2',
			)
		),
		'file_url' => Array (
#			'exclude' => 1,
			'label' => 'LLL:EXT:sevenpack/locallang_db.xml:tx_sevenpack_references_file_url',
			'config' => Array (
				'type' => 'input',
				'size' => '48',
				'max' => '255',
				'checkbox' => '0',
				'eval' => 'trim',
				'wizards' => Array(
					'_PADDING' => 2,
					'link' => Array(
						'type' => 'popup',
						'title' => 'Link',
						'icon' => 'link_popup.gif',
						'script' => 'browse_links.php?mode=wizard',
						'JSopenParams' => 'height=300,width=500,status=0,menubar=0,scrollbars=1'
					),
				),
			)
		),
		'web_url' => Array (
#			'exclude' => 1,
			'label' => 'LLL:EXT:sevenpack/locallang_db.xml:tx_sevenpack_references_web_url',
			'config' => Array (
				'type' => 'input',
				'size' => '48',
				'max' => '255',
				'checkbox' => '0',
				'wizards' => Array(
					'_PADDING' => 2,
					'link' => Array(
						'type' => 'popup',
						'title' => 'Link',
						'icon' => 'link_popup.gif',
						'script' => 'browse_links.php?mode=wizard',
						'JSopenParams' => 'height=300,width=500,status=0,menubar=0,scrollbars=1'
					),
				),
			)
		),
		'web_url2' => Array (
#			'exclude' => 1,
			'label' => 'LLL:EXT:sevenpack/locallang_db.xml:tx_sevenpack_references_web_url2',
			'config' => Array (
				'type' => 'input',
				'size' => '48',
				'max' => '255',
				'checkbox' => '0',
				'wizards' => Array(
					'_PADDING' => 2,
					'link' => Array(
						'type' => 'popup',
						'title' => 'Link',
						'icon' => 'link_popup.gif',
						'script' => 'browse_links.php?mode=wizard',
						'JSopenParams' => 'height=300,width=500,status=0,menubar=0,scrollbars=1'
					),
				),
			)
		),
		'misc' => Array (
#			'exclude' => 1,
			'label' => 'LLL:EXT:sevenpack/locallang_db.xml:tx_sevenpack_references_misc',
			'config' => Array (
				'type' => 'input',
				'size' => '48',
				'max' => '255',
				'eval' => 'trim',
			)
		),
		'misc2' => Array (
#			'exclude' => 1,
			'label' => 'LLL:EXT:sevenpack/locallang_db.xml:tx_sevenpack_references_misc2',
			'config' => Array (
				'type' => 'input',
				'size' => '48',
				'max' => '255',
				'eval' => 'trim',
			)
		),
		'editor' => Array (
#			'exclude' => 1,
			'label' => 'LLL:EXT:sevenpack/locallang_db.xml:tx_sevenpack_references_editor',
			'config' => Array (
				'type' => 'input',
				'size' => '48',
				'max' => '255',
				'eval' => 'trim',
			)
		),
		'publisher' => Array (
#			'exclude' => 1,
			'label' => 'LLL:EXT:sevenpack/locallang_db.xml:tx_sevenpack_references_publisher',
			'config' => Array (
				'type' => 'input',
				'size' => '48',
				'max' => '255',
				'eval' => 'trim',
			)
		),
		'address' => Array (
#			'exclude' => 1,
			'label' => 'LLL:EXT:sevenpack/locallang_db.xml:tx_sevenpack_references_address',
			'config' => Array (
				'type' => 'input',
				'size' => '48',
				'max' => '255',
				'eval' => 'trim',
			)
		),
		'howpublished' => Array (
#			'exclude' => 1,
			'label' => 'LLL:EXT:sevenpack/locallang_db.xml:tx_sevenpack_references_howpublished',
			'config' => Array (
				'type' => 'input',
				'size' => '48',
				'max' => '255',
				'eval' => 'trim',
			)
		),
		'series' => Array (
#			'exclude' => 1,
			'label' => 'LLL:EXT:sevenpack/locallang_db.xml:tx_sevenpack_references_series',
			'config' => Array (
				'type' => 'input',
				'size' => '48',
				'max' => '255',
				'eval' => 'trim',
			)
		),
		'edition' => Array (
#			'exclude' => 1,
			'label' => 'LLL:EXT:sevenpack/locallang_db.xml:tx_sevenpack_references_edition',
			'config' => Array (
				'type' => 'input',
				'size' => '48',
				'max' => '255',
				'eval' => 'trim',
			)
		),
		'chapter' => Array (
#			'exclude' => 1,
			'label' => 'LLL:EXT:sevenpack/locallang_db.xml:tx_sevenpack_references_chapter',
			'config' => Array (
				'type' => 'input',
				'size' => '48',
				'max' => '255',
				'eval' => 'trim',
			)
		),
		'booktitle' => Array (
#			'exclude' => 1,
			'label' => 'LLL:EXT:sevenpack/locallang_db.xml:tx_sevenpack_references_booktitle',
			'config' => Array (
				'type' => 'text',
				'cols' => '48',
				'rows' => '2',
				'eval' => 'trim',
			)
		),
		'school' => Array (
#			'exclude' => 1,
			'label' => 'LLL:EXT:sevenpack/locallang_db.xml:tx_sevenpack_references_school',
			'config' => Array (
				'type' => 'input',
				'size' => '48',
				'max' => '255',
				'eval' => 'trim',
			)
		),
		'institute' => Array (
#			'exclude' => 1,
			'label' => 'LLL:EXT:sevenpack/locallang_db.xml:tx_sevenpack_references_institute',
			'config' => Array (
				'type' => 'input',
				'size' => '48',
				'max' => '255',
				'eval' => 'trim',
			)
		),
		'organization' => Array (
#			'exclude' => 1,
			'label' => 'LLL:EXT:sevenpack/locallang_db.xml:tx_sevenpack_references_organization',
			'config' => Array (
				'type' => 'input',
				'size' => '48',
				'max' => '255',
				'eval' => 'trim',
			)
		),
		'institution' => Array (
#			'exclude' => 1,
			'label' => 'LLL:EXT:sevenpack/locallang_db.xml:tx_sevenpack_references_institution',
			'config' => Array (
				'type' => 'input',
				'size' => '48',
				'max' => '255',
				'eval' => 'trim',
			)
		),
		'event_name' => Array (
#			'exclude' => 1,
			'label' => 'LLL:EXT:sevenpack/locallang_db.xml:tx_sevenpack_references_event_name',
			'config' => Array (
				'type' => 'input',
				'size' => '48',
				'max' => '255',
				'eval' => 'trim',
			)
		),
		'event_place' => Array (
#			'exclude' => 1,
			'label' => 'LLL:EXT:sevenpack/locallang_db.xml:tx_sevenpack_references_event_place',
			'config' => Array (
				'type' => 'input',
				'size' => '48',
				'max' => '255',
				'eval' => 'trim',
			)
		),
		'event_date' => Array (
#			'exclude' => 1,
			'label' => 'LLL:EXT:sevenpack/locallang_db.xml:tx_sevenpack_references_event_date',
			'config' => Array (
				'type' => 'input',
				'size' => '48',
				'max' => '255',
				'eval' => 'trim',
			)
		),
		'state' => Array (
#			'exclude' => 1,
			'label' => 'LLL:EXT:sevenpack/locallang_db.xml:tx_sevenpack_references_state',
			'config' => Array (
				'type' => 'select',
				'items' => Array (
					Array('LLL:EXT:sevenpack/locallang_db.xml:tx_sevenpack_references_state_I_0', '0'),
					Array('LLL:EXT:sevenpack/locallang_db.xml:tx_sevenpack_references_state_I_1', '1'),
					Array('LLL:EXT:sevenpack/locallang_db.xml:tx_sevenpack_references_state_I_2', '2'),
					Array('LLL:EXT:sevenpack/locallang_db.xml:tx_sevenpack_references_state_I_3', '3'),
					Array('LLL:EXT:sevenpack/locallang_db.xml:tx_sevenpack_references_state_I_4', '4'),
				),
				'size' => 1,
				'maxitems' => 1,
			)
		),
		'type' => Array (
#			'exclude' => 1,
			'label' => 'LLL:EXT:sevenpack/locallang_db.xml:tx_sevenpack_references_type',
			'config' => Array (
				'type' => 'input',
				'size' => '48',
				'max' => '255',
				'eval' => 'trim',
			)
		),
		'language' => Array (
#			'exclude' => 1,
			'label' => 'LLL:EXT:sevenpack/locallang_db.xml:tx_sevenpack_references_language',
			'config' => Array (
				'type' => 'input',
				'size' => '48',
				'max' => '255',
				'eval' => 'trim',
			)
		),
		'ISBN' => Array (
#			'exclude' => 1,
			'label' => 'LLL:EXT:sevenpack/locallang_db.xml:tx_sevenpack_references_ISBN',
			'config' => Array (
				'type' => 'input',
				'size' => '48',
				'max' => '255',
				'eval' => 'trim',
			)
		),
		'ISSN' => Array (
#			'exclude' => 1,
			'label' => 'LLL:EXT:sevenpack/locallang_db.xml:tx_sevenpack_references_ISSN',
			'config' => Array (
				'type' => 'input',
				'size' => '48',
				'max' => '255',
				'eval' => 'trim',
			)
		),
		'DOI' => Array (
#			'exclude' => 1,
			'label' => 'LLL:EXT:sevenpack/locallang_db.xml:tx_sevenpack_references_DOI',
			'config' => Array (
				'type' => 'input',
				'size' => '48',
				'max' => '255',
				'eval' => 'trim',
			)
		),
		'extern' => Array (
#			'exclude' => 1,
			'label' => 'LLL:EXT:sevenpack/locallang_db.xml:tx_sevenpack_references_extern',
			'config' => Array (
				'type'    => 'check',
				'default' => '0'
			)
		),
		'reviewed' => Array (
#			'exclude' => 1,
			'label' => 'LLL:EXT:sevenpack/locallang_db.xml:tx_sevenpack_references_reviewed',
			'config' => Array (
				'type'    => 'check',
				'default' => '0'
			)
		),
		'in_library' => Array (
#			'exclude' => 1,
			'label' => 'LLL:EXT:sevenpack/locallang_db.xml:tx_sevenpack_references_in_library',
			'config' => Array (
				'type'    => 'check',
				'default' => '0'
			)
		),
		'borrowed_by' => Array (
#			'exclude' => 1,
			'label' => 'LLL:EXT:sevenpack/locallang_db.xml:tx_sevenpack_references_borrowed_by',
			'config' => Array (
				'type' => 'input',
				'size' => '48',
				'max' => '255',
				'eval' => 'trim',
			)
		),
	),
	'types' => array (
		'0' => array ( 'showitem' => 'hidden;;1, bibtype, citeid, title;;;;2-2-2, journal;;;;3-3-3, year, month, day, volume, number, number2, pages, abstract, affiliation, note, annotation, keywords, tags, file_url, web_url, web_url2, misc, misc2, editor, publisher, address, howpublished, series,  edition, chapter, booktitle, school, institute, organization, institution, event_name, event_place, event_date, state, type, language, ISBN, ISSN, extern, reviewed, in_library, borrowed_by' )
	),
);


$TCA['tx_sevenpack_authors'] = array (
	'ctrl' => $TCA['tx_sevenpack_authors']['ctrl'],
	'interface' => array (
		'showRecordFieldList' => $TCA['tx_sevenpack_authors']['feInterface']['fe_admin_fieldList']
	),
	'feInterface' => $TCA['tx_sevenpack_authors']['feInterface'],
	'columns' => array (
		'surname' => array (
#			'exclude' => 1,
			'label' => 'LLL:EXT:sevenpack/locallang_db.xml:tx_sevenpack_authors_surname',
			'config' => Array (
				'type' => 'input',
				'size' => '48',
				'max' => '255',
				'eval' => 'trim,required',
			)
		),
		'forename' => array (		
#			'exclude' => 1,
			'label' => 'LLL:EXT:sevenpack/locallang_db.xml:tx_sevenpack_authors_forename',
			'config' => Array (
				'type' => 'input',
				'size' => '48',
				'max' => '255',
				'eval' => 'trim',
			)
		),
		'url' => Array (
#			'exclude' => 1,
			'label' => 'LLL:EXT:sevenpack/locallang_db.xml:tx_sevenpack_authors_url',
			'config' => Array (
				'type' => 'input',
				'size' => '48',
				'max' => '255',
				'checkbox' => '0',
				'wizards' => Array(
					'_PADDING' => 2,
					'link' => Array(
						'type' => 'popup',
						'title' => 'Link',
						'icon' => 'link_popup.gif',
						'script' => 'browse_links.php?mode=wizard',
						'JSopenParams' => 'height=300,width=500,status=0,menubar=0,scrollbars=1'
					)
				)
			)
		),
		'fe_user_id' => Array (
#			'exclude' => 1,
			'label' => 'LLL:EXT:sevenpack/locallang_db.xml:tx_sevenpack_authors_fe_user_id',
			'config' => Array (
				'type' => 'group',
				'size' => 1,
				'internal_type' => 'db',
				'allowed' => 'fe_users',
				'minitems' => 0,
				'maxitems' => 1,
			)
		),
	),
	'types' => array (
		'0' => array ( 'showitem' => 'surname,forename,url,fe_user_id' )
	),
);


$TCA['tx_sevenpack_authorships'] = array (
	'ctrl' => $TCA['tx_sevenpack_authorships']['ctrl'],
	'interface' => array (
		'showRecordFieldList' => $TCA['tx_sevenpack_authorships']['feInterface']['fe_admin_fieldList']
	),
	'feInterface' => $TCA['tx_pmeightpack_authorships']['feInterface'],
	'columns' => array (
		'pub_id' => array (
#      "exclude" => 1,
			'label' => 'LLL:EXT:sevenpack/locallang_db.xml:tx_sevenpack_authorships_pub_id',
			'config' => array (
				'type' => 'group',
				'internal_type' => 'db',
				'allowed' => 'tx_sevenpack_references',
				'size' => 1,
				'minitems' => 1,
				'maxitems' => 1,
				'selectedListStyle' => 'width:320px;',
				'wizards' => array (
					'_VALIGN' => 'top',
					'ajax_search' => array (
						'type' => 'userFunc',
						'userFunc' => 'tx_ajaxgroupsearch_client->renderAjaxSearch',
						'params' => array (
							'client' => array ( 'startLength' => 2 ),
							'wrapStyle' => 'z-index:80;',
							'inputStyle' => 'width:200px;',
							'itemListStyle' => 'width:320px;',
							'tables' => array (
								'tx_sevenpack_references' => array (
									'searchBySQL' => array (
									'fields' => 'r.title, r.uid',
									'tables' => 'tx_sevenpack_references AS r',
									'where' => 'r.title LIKE "%###SEARCHWORD###%" AND r.deleted=0 AND r.hidden=0',
									'group_by' => '',
									'order_by' => 'r.title DESC',
									'limit' => '10'
									)
								)
							)
						)
					)
				)
			)
		),
		'author_id' => array (
#      "exclude" => 1,
			'label' => 'LLL:EXT:sevenpack/locallang_db.xml:tx_sevenpack_authorships_author_id',
			'config' => array (
				'type' => 'group',
				'internal_type' => 'db',
				'allowed' => 'tx_sevenpack_authors',
				'size' => 1,
				'minitems' => 1,
				'maxitems' => 1,
				'selectedListStyle' => 'width:320px;',
				'wizards' => array(
					'_VALIGN' => 'top',
					'ajax_search' => array(
						'type' => 'userFunc',
						'userFunc' => 'tx_ajaxgroupsearch_client->renderAjaxSearch',
						'params' => array(
							'client' => array('startLength'=>2),
							'wrapStyle' => 'z-index:80;',
							'inputStyle' => 'width:200px;',
							'itemListStyle' => 'width:320px;',
							'tables' => array(
								'tx_sevenpack_authors' => array(
									'searchBySQL' => array(
									'fields' => 'a.surname, a.uid',
									'tables' => 'tx_sevenpack_authors AS a',
									'where' => 'a.surname LIKE "%###SEARCHWORD###%" AND a.deleted=0 AND a.hidden=0',
									'group_by' => '',
									'order_by' => 'a.surname DESC, a.forename DESC',
									'limit' => '10'
									)
								)
							)
						)
					)
				)
			)
		),
		'sorting' => array (
#			'exclude' => 1,
			'label' => 'LLL:EXT:sevenpack/locallang_db.xml:tx_sevenpack_authorships_sorting',
			'config' => Array (
				'type' => 'input',
				'size' => '10',
				'max' => '255',
				'eval' => 'int',
			)
		),
	),
	'types' => array (
		'0' => array ( 'showitem' => 'pub_id,author_id,sorting' )
	),
);


?>
